src/site/controllers/WebController.php

```diff
@@ -89,15 +84,13 @@ protected static function getExtraParametersSpec(
     return shape('required' => ImmVector {}, 'optional' => ImmVector {});
   }
 
-  abstract public function getResponseAsync(
-    ResponseInterface $response,
-  ): Awaitable<ResponseInterface>;
+  abstract public function getResponseAsync(): Awaitable<Message\IResponse>;
```

Comment:
we don't pass a response to the controller anymore as we don't actually need to, previously it was required as the response body is only writable, in nuxed case, its both readable and writeable, meaning the controller can write anything, and we can read from it without knowing where the body is located ( in case its a file ). we know its readable, we can read from it at the end ( done int the emitter )